### ISSUE TITLE
tensor data_format

### ISSUE THREAD

**Issue by vikyzeng** [wchen342/SketchyGAN#20]
Dear Dr. Chen. When I trained on two GPUs, there is always a problem about data_format.

InvalidArgumentError (see above for traceback): Conv2DCustomBackpropInputOp only supports NHWC.
	 [[node GPU_0/gradients/GPU_0/discriminator_1/Conv_1/Conv2D_grad/Conv2DBackpropInput (defined at ./src_single/graph_single.py:26)  = Conv2DBackpropInput[T=DT_FLOAT, data_format="NCHW", dilations=[1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 1], use_cudnn_on_gpu=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](GPU_0/gradients/GPU_0/discriminator_1/Conv_1/Conv2D_grad/ShapeN, GPU_0/discriminator_1/Conv_1/Reshape_1, GPU_0/gradients/GPU_0/discriminator_1/Conv_1/add_grad/tuple/control_dependency)]]

In config.py, you note that 'data_format='NCHW' should not changed. I wonder why the problem is raised? Looking forward to your apply! Thank you very much!

**Comment by wchen342** [comment]
It has something to do with your tensorflow and cuda. What is your system
info (system version, tf version, etc.)?
Some guys reported this happens on Windows 10 but disappeared after
switching to Ubuntu.

**Comment by vikyzeng** [comment]
Thank you for apply! It worked on Ubuntu. Is it feasible to change the data_format to NHWC and delete those assert in code to fit my environment?

**Comment by wchen342** [comment]
If you know how to change the code since there are a bunch of places need
to be changed, like transposes.

**Comment by vikyzeng** [comment]
I will give it a try! By the way, I have sent my sftp info to your school email to get the augmented dataset. Have you received it? Thank you very much in advance!

**Comment by wchen342** [comment]
Yeah, I will upload later.

Edit: Can you use another service provider? I seem to have problem with connection speed.

**Comment by vikyzeng** [comment]
> Yeah, I will upload later.
> Edit: Can you use another service provider? I seem to have problem with connection speed.

The server in china may have some problem when connecting to extranet. Is it possible to upload the dataset to Google Driver or other Cloud Driver temporarily? Sorry for bringing you inconvenience！
And can I know the size of the dataset?

**Comment by vikyzeng** [comment]
Can I know the size of the dataset?

**Comment by wchen342** [comment]
The whole dataset is 170+G and Google Drive has only 10-15G. I will suggest you setup a vps like GCP or DO temporarily. It's not free but it should work better.

**Comment by vikyzeng** [comment]
I have 2T free space in Baidu Cloud Disk. Is it convenient to use Baidu Cloud Disk in your country? I can offer my account and password to you to upload the dataset. If it is convenient to install the Baidu Cloud Disk and the upload speed is fine, it would be the easiest way. If it has some problems by this way, I will try to set up a vps.

Sorry for bothering you! Looking forward to your reply. 

----- 原始邮件 -----
发件人: "wchen342" <notifications@github.com>
收件人: "wchen342/SketchyGAN" <SketchyGAN@noreply.github.com>
抄送: "vikyzeng" <zengyihan@sjtu.edu.cn>, "Author" <author@noreply.github.com>
发送时间: 星期四, 2019年 1 月 17日 下午 3:07:43
主题: Re: [wchen342/SketchyGAN] tensor data_format (#20)

The whole dataset is 170+G and Google Drive has only 10-15G. I will suggest you setup a vps like GCP or DO temporarily. It's not free but it should work better.

**Comment by wchen342** [comment]
I can try that, it will probably be better.

**Comment by vikyzeng** [comment]
> I can try that, it will probably be better.

Dear Dr.Chen. I have sent my Baidu Cloud Disk account to your email. Have you received it?  Thank you for your kind cooperation!
By the way, I try to make tfrecord file for my own edge images. I extract edge images through batch_hed.py - PostprocessHED.m. But the dist_map turn to be whole black. Is this because the edge line is too thin?  Have you encountered such a problem?
Looking forward to your reply!

**Comment by 15702435639** [comment]
Dear vikyzeng,Like you, I am also making my own dataset.But I don't’ know how to convert image format to TFRecord format.Could you tell me the specific detail or code?Thanks very much.

**Comment by 15702435639** [comment]
Dear vikyzeng,Like you, I am also making my own dataset.But I don't’ know how to convert image format to TFRecord format.Could you tell me the specific detail or code?Thanks very much.

**Comment by wchen342** [comment]
The scripts are uploaded to `data_processing`.
